### PR TITLE
Add bedwars barebones

### PR DESCRIPTION
**Pull request by PancakeTAS**
- [x] Tool/Armor tiers (needs testing)
- [x] Tnt/Fireball mechanics
- [x] Item shop
- [x] Resource spawners
- [ ] Progressive game updates (spawner upgrades, sudden death)
- [ ] Teams and beds
- [ ] Scoreboard
- [ ] Team upgrades
- [ ] Block placement restrictions (missing in-base restrictions)
- [ ] Gamemode variations (deathswap, rush)
- [ ] Testing, tweaking, polishing, etc.

QOL:
- [ ] Resource spawner limit
- [ ] Adjust tnt/fireball yield
- [ ] Unbreakable tools
- [ ] Various game events such as food

### CODE REVIEW

**Review by ScribbleTAS — APPROVED** [review]

**Comment by PancakeTAS** [comment]
would not recommend changes since last commit instead full pr for review

**Review by ScribbleTAS — APPROVED** [review]
Nice

**Review by ScribbleTAS — APPROVED** [review]

File: Gameserver/src/bedwars/java/com/minecrafttas/tasbattle/bedwars/components/InventoryManagement.java [hunk 1]
```diff
@@ -71,14 +71,15 @@ public void onTick(ServerTickStartEvent e) {
 		for (var p : this.players) {
 			var inv = p.getInventory();
 			
-			// verify at least one sword in inventory
-			if (!inv.contains(Material.WOODEN_SWORD) && !inv.contains(Material.STONE_SWORD) && !inv.contains(Material.IRON_SWORD) && !inv.contains(Material.DIAMOND_SWORD))
+			// verify the correct sword is in the inventory
```
Comment (ScribbleTAS) [inline Gameserver/src/bedwars/java/com/minecrafttas/tasbattle/bedwars/components/InventoryManagement.java:74]: As a sidenote, all comments in this file are missing a "that"
```
// verify that the correct sword is in the inventory
```
Not a dealbreaker here but it sounds weird when reading through them

**Review by ScribbleTAS — APPROVED** [review]

File: Gameserver/src/bedwars/java/com/minecrafttas/tasbattle/bedwars/components/InventoryManagement.java (hunk 1 above)
Comment (PancakeTAS) [reply Gameserver/src/bedwars/java/com/minecrafttas/tasbattle/bedwars/components/InventoryManagement.java:74]: it's OPTIMAL! lol

**Comment by PancakeTAS** [comment]
You know what Scribble? I can't be fucked to program this gamemode anymore.. I would just leave this pull request until I want to work on it again, but I changed some crucial other part of the common library making it impossible to develop other gamemodes without the merge conflict pain. Leave your review, I'll merge this pull request...

**Review by ScribbleTAS — APPROVED** [review]

**Pull request merged**